### PR TITLE
diamond_dust to field_system_weather

### PR DESCRIPTION
**Pull request by red031000**

### CODE REVIEW

File: src/unk_02055418.c [hunk 1]
```diff
@@ -24,7 +24,7 @@
 void FieldSys_StartBugContestTimer(FieldSystem* fsys) {
     RTCDate date;
     RTCTime time;
-    SYSINFO_RTC* sysinfo = Save_SysInfo_RTC_Get(fsys->savedata);
```
Comment (adrienntindall) [inline src/unk_02055418.c:27]: Pointer notation in this file

File: src/sav_system_info.c [hunk 2]
```diff
@@ -102,7 +102,7 @@ void Save_SysInfo_RTC_SubField34(SYSINFO_RTC* rtc_info, u32 a1) {
     rtc_info->unk34 -= a1;
 }
 
-void SysInfoRTC_HandleContinueOnNewConsole(SYSINFO_RTC* rtc_info) {
```
Comment (adrienntindall) [inline src/sav_system_info.c:105]: Pointer notation in this file

File: src/field_system_weather.c [hunk 3]
```diff
@@ -15,19 +15,18 @@ const MonthDay DiamondDustDates[] = {
     { DECEMBER, 31 },
 }; //_020FA09C
 
-u32 Fsys_GetWeather_HandleDiamondDust(FieldSystem* fsys, u32 mapId) {
+u32 FieldSystem_GetWeather_HandleDiamondDust(FieldSystem *fieldSystem, u32 mapId) {
```
Comment (adrienntindall) [inline src/field_system_weather.c:18]: why did you change fsys to fieldSystem?

File: main.lsf [hunk 4]
```diff
@@ -155,7 +155,7 @@ Static main
 	Object src/map_events.o
 	Object src/save_local_field_data.o
 	Object asm/unk_0203BA5C.o
-	Object src/diamond_dust.o
+	Object src/field_system_weather.o
```
Comment (adrienntindall) [inline main.lsf:158]: I'm ok with this so long as we're sure this is the only file related to field weather (ie it'd make more sense to use the field_system_weather name for the file that handles the gfx of weather in the field)

**Review by adrienntindall — CHANGES_REQUESTED** [review]

File: src/field_system_weather.c (hunk 3 above)
Comment (red031000) [reply src/field_system_weather.c:18]: consistency with diamond, literally no other reason

File: main.lsf (hunk 4 above)
Comment (red031000) [reply main.lsf:158]: if you have a better name suggestion, then I can take it on board

File: src/unk_02055418.c (hunk 1 above)
Comment (red031000) [reply src/unk_02055418.c:27]: out of scope - will be addressed in another PR, when I get to the file

File: src/sav_system_info.c (hunk 2 above)
Comment (red031000) [reply src/sav_system_info.c:105]: out of scope - will be addressed in another PR, when I get to the file

File: src/field_system_weather.c (hunk 3 above)
Comment (adrienntindall) [reply src/field_system_weather.c:18]: we use fsys here bucko

Comment (red031000) [reply src/field_system_weather.c:18]: fsys is dumb, fieldsystem is better, the struct is named fieldsystem, unless you want me to change that to struct FSys :)

File: main.lsf (hunk 4 above)
Comment (adrienntindall) [reply main.lsf:158]: lets hold off till we decomp more of the weather related files and see if it still works or not? otherwise this literally only affects diamond dust and I'm already struggling with conflicting names 1/3 of the way through battle code so gonna try to avoid that in the field

File: src/field_system_weather.c (hunk 3 above)
Comment (adrienntindall) [reply src/field_system_weather.c:18]: you're dumb >=(

File: main.lsf (hunk 4 above)
Comment (red031000) [reply main.lsf:158]: I'd rather not name it diamond_dust.c as the equivalent file in diamond deals with more than just diamond dust

Comment (adrienntindall) [reply main.lsf:158]: The main thing Im concerned about is how much different weather is handled between the two games, since a lot of it was stripped in HGSS

File: src/unk_02055418.c (hunk 1 above)
Comment (adrienntindall) [reply src/unk_02055418.c:27]: it's literally a cntrl + f and replace smh

File: include/field_system_weather.h
Comment (adrienntindall) [inline include/field_system_weather.h:1]: File name not changed yet

**Review by adrienntindall — CHANGES_REQUESTED** [review]

**Review by adrienntindall — APPROVED** [review]
